index.js

```diff
@@ -1,45 +1,58 @@
-import express from "express";
 import geohash from "ngeohash";
 import {
   uniqueNamesGenerator,
   adjectives,
   colors,
   animals,
 } from "unique-names-generator";
+import express from "express";
+import { createServer } from "http";
+import { Server } from "socket.io";
+
+const app = express();
+const httpServer = createServer(app);
+const io = new Server(httpServer);
 
 /**
  * @type {Object.<string, Set>}
  */
 const grids = {};
-const app = express();
 
 app.use(express.json());
 app.use(express.static("public"));
 
-app.post("/api/initial", (req, res) => {
-  const { lat, lon } = req.body;
-  // geohash of length 5 is <= 4.89km X 4.89km
-  const hash = geohash.encode(lat, lon, 5);
+io.on("connection", (socket) => {
+  const handleSignin = (args) => {
+    const { lat, lon } = args || {};
+
+    if (!lat || !lon) {
```

Comment:
I'm not sure what should we do here. Do we throw an error and let the client listen to the ?event `connect_error `?

Reply:
You can emit the `signin` event here as well, just return an object with error instead `{ error: "..." }` instead of `{ hash, username }` and let the client check that the signin event has no error prop before proceeding.